src/services/room-service.spec.js

```diff
@@ -276,6 +302,33 @@ describe('room-service', () => {
         const invitationFromDb = await db.roomInvitations.findOne({ _id: invitation._id });
         expect(invitationFromDb).toBeNull();
       });
+
+      describe('and the user gets invited a second time', () => {
+        let exisingMemberJoinDate;
+
+        beforeEach(async () => {
+          ({ invitation } = await sut.createOrUpdateInvitation({ roomId: testRoom._id, email: otherUser.email, user: myUser }));
+
+          const roomFromDb = await db.rooms.findOne({ _id: testRoom._id });
+          exisingMemberJoinDate = roomFromDb.members[0].joinedOn;
+
+          await sut.confirmInvitation({ token: invitation.token, user: otherUser });
+        });
+
+        it('should not add the user user a second time', async () => {
+          const roomFromDb = await db.rooms.findOne({ _id: testRoom._id });
+          expect(roomFromDb.members).toHaveLength(1);
+          expect(roomFromDb.members[0]).toEqual({
+            userId: otherUser._id,
+            joinedOn: exisingMemberJoinDate
+          });
```

Comment:
You can merge the expectations into one:

```
expect(roomFromDb.members).toEqual([{
    userId: otherUser._id,
    joinedOn: exisingMemberJoinDate
}]);
```

Reply:
Adjusted both here and in older code.